src/test/booking.test.js

```diff
@@ -0,0 +1,316 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import sinon from 'sinon';
+import sinonChai from 'sinon-chai';
+import faker from 'faker';
+import server from '../index';
+import { BookingController, AuthController } from '../controllers';
+import { newSupplier, newFacility } from './dummies';
+
+chai.use(chaiHttp);
+chai.use(sinonChai);
+
+const { supplierSignup } = AuthController;
+
+describe('Booking Test', () => {
+  let roomId;
+  let adminToken;
+  before(async () => {
+    const supplierData = { ...newSupplier, email: faker.internet.email() };
+    const supReq = {
+      body: { ...supplierData }
+    };
+
+    const res = {
+      cookie() { return this; },
+      status() { return this; },
+      json(obj) { return obj; }
+    };
+
+    const supplierResponse = await supplierSignup(supReq, res);
+    adminToken = supplierResponse.data.user.token;
+
+    const roomResponse = await chai
+      .request(server)
+      .post('/api/facility/supplier')
+      .set('Authorization', `Bearer ${adminToken}`)
+      .send(newFacility);
+    roomId = roomResponse.body.data.rooms[0].id;
+  });
+
+  afterEach(() => sinon.restore());
+  describe('Booking Route Endpoints', () => {
+    it('should successfully create an accommodation booking', async () => {
+      const booking = {
+        checkIn: '2030-12-20',
+        checkOut: '2030-12-30',
+        userId: 1,
+        roomId
+      };
+
+      const response = await chai
+        .request(server)
+        .post('/api/booking/accommodation')
+        .set('Authorization', `Bearer ${adminToken}`)
+        .send(booking);
+      expect(response).to.have.status(201);
+      expect(response.body.data).to.be.a('object');
```

Comment:
Can we check the contents of the object to ensure we have the correct object data? Check out the `toMatchObject` method

Reply:
Couldn't find the toMatchObject method in chai's assertion library

Reply:
My bad, I assumed the test runner is Jest; that method is for Jest. For Mocha and Chai, you can use `include` or `deep include`

Reply:
Have been implemented